pkg/clients/clients.go

```diff
@@ -118,23 +117,18 @@ func NewDynamicClientSet(clientConfig *rest.Config) (*DynamicClientSet, error) {
 	}, nil
 }
 
-func (dcs *DynamicClientSet) ResourceClient(gvk schema.GroupVersionKind, namespace string) (dynamic.ResourceInterface, error) {
+func (dcs *DynamicClientSet) ResourceClient(gvk schema.GroupVersionKind, namespace string,
+) (dynamic.ResourceInterface, error) {
 	m, err := dcs.RESTMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 	if err != nil {
 		// If the REST mapping failed, try refreshing the cache and remapping before giving up.
 		// This can occur if a CRD is being registered from another resource.
 		dcs.DiscoveryClientCached.Invalidate()
 		dcs.RESTMapper.Reset()
 
-		// TODO(lblackstone): Ensure that retry behavior is predictable.
-		err := retry.SleepingRetry(
-			func(i uint) (err error) {
-				m, err = dcs.RESTMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-				return
-			}).
-			WithMaxRetries(5).
-			WithBackoffFactor(2).
-			Do(meta.IsNoMatchError)
+		time.Sleep(2 * time.Second)
```

Comment:
Do we actually need to sleep and retry here at all? It _seems_ like the invalidation and reset occur (a) synchronously, and (b) immediately?

Reply:
Ah, you're right! I hadn't looked closely at the `Invalidate` method before. While digging, I also discovered that the `Reset` command calls `Invalidate`, so I'll remove the redundancy there.